client/src/Components/Card/FuelLeftCard.js

```diff
@@ -29,8 +30,34 @@ class FuelLeftCard extends Component {
       this.kmSinceLastRefill = this.odometer - this.lastRefillOdo;
       this.kmSinceLastRefillRounded = Math.floor(this.kmSinceLastRefill);
 
+
+      const refillData = this.props.refillData
```

Comment:
I would put this calculation logic on the server side - it doesn't need to be recalculated every time the component renders. Also we should write unit tests for this.